free_fleet_server_ros2 build fails for Galactic

## Bug report

### Required information:
-ROS2 Galactic

### Description of the bug
Some API changes break the free_fleet_server_ros2 build:
```
--- stderr: free_fleet_server_ros2
In file included from /opt/rmf/src/free_fleet/free_fleet_server_ros2/src/main.cpp:22:
/opt/rmf/src/free_fleet/free_fleet_server_ros2/src/ServerNode.hpp:110:11: error: ‘callback_group’ in namespace ‘rclcpp’ does not name a type; did you mean ‘CallbackGroup’?
  110 |   rclcpp::callback_group::CallbackGroup::SharedPtr update_state_callback_group;
      |           ^~~~~~~~~~~~~~
      |           CallbackGroup
/opt/rmf/src/free_fleet/free_fleet_server_ros2/src/ServerNode.hpp:123:11: error: ‘callback_group’ in namespace ‘rclcpp’ does not name a type; did you mean ‘CallbackGroup’?
  123 |   rclcpp::callback_group::CallbackGroup::SharedPtr
      |           ^~~~~~~~~~~~~~
      |           CallbackGroup
/opt/rmf/src/free_fleet/free_fleet_server_ros2/src/main.cpp: In function ‘int main(int, char**)’:
/opt/rmf/src/free_fleet/free_fleet_server_ros2/src/main.cpp:39:15: error: ‘rclcpp::executor’ has not been declared
   39 |       rclcpp::executor::ExecutorArgs(), 2};
      |               ^~~~~~~~
/opt/rmf/src/free_fleet/free_fleet_server_ros2/src/main.cpp:39:42: error: no matching function for call to ‘rclcpp::executors::MultiThreadedExecutor::MultiThreadedExecutor(<brace-enclosed initializer list>)’
   39 |       rclcpp::executor::ExecutorArgs(), 2};
      |                                          ^
In file included from /opt/ros/galactic/include/rclcpp/executors.hpp:21,
                 from /opt/ros/galactic/include/rclcpp/rclcpp.hpp:156,
                 from /opt/rmf/src/free_fleet/free_fleet_server_ros2/src/main.cpp:20:
/opt/ros/galactic/include/rclcpp/executors/multi_threaded_executor.hpp:56:3: note: candidate: ‘rclcpp::executors::MultiThreadedExecutor::MultiThreadedExecutor(const rclcpp::ExecutorOptions&, size_t, bool, std::chrono::nanoseconds)’
   56 |   MultiThreadedExecutor(
      |   ^~~~~~~~~~~~~~~~~~~~~
/opt/ros/galactic/include/rclcpp/executors/multi_threaded_executor.hpp:56:3: note:   conversion of argument 1 would be ill-formed:
make[2]: *** [CMakeFiles/free_fleet_server_ros2.dir/build.make:63: CMakeFiles/free_fleet_server_ros2.dir/src/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/free_fleet_server_ros2.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
---
Failed   <<< free_fleet_server_ros2 [3.93s, exited with code 2]

```

--- a/free_fleet_server_ros2/src/ServerNode.cpp
+++ b/free_fleet_server_ros2/src/ServerNode.cpp
@@ -53,7 +53,7 @@ ServerNode::SharedPtr ServerNode::make(
       break;
     RCLCPP_INFO(
         server_node->get_logger(), "waiting for configuration parameters.");
-    
+
     end_time = std::chrono::steady_clock::now();
     std::this_thread::sleep_for(std::chrono::milliseconds(1000));
   }
@@ -67,7 +67,7 @@ ServerNode::SharedPtr ServerNode::make(
   server_node->print_config();
 
   // Starting the free fleet server
-  ServerConfig server_config = 
+  ServerConfig server_config =
       server_node->server_node_config.get_server_config();
   Server::SharedPtr server = Server::make(server_config);
   if (!server)
@@ -84,7 +84,7 @@ ServerNode::~ServerNode()
 {}
 
 ServerNode::ServerNode(
-    const ServerNodeConfig& _config, 
+    const ServerNodeConfig& _config,
     const rclcpp::NodeOptions& _node_options) :
   Node(_config.fleet_name + "_node", _node_options),
   server_node_config(_config)
@@ -102,19 +102,19 @@ void ServerNode::setup_config()
   get_parameter("mode_request_topic", server_node_config.mode_request_topic);
   get_parameter("path_request_topic", server_node_config.path_request_topic);
   get_parameter(
-      "destination_request_topic", 
+      "destination_request_topic",
       server_node_config.destination_request_topic);
   get_parameter("dds_domain", server_node_config.dds_domain);
-  get_parameter("dds_robot_state_topic", 
+  get_parameter("dds_robot_state_topic",
       server_node_config.dds_robot_state_topic);
-  get_parameter("dds_mode_request_topic", 
+  get_parameter("dds_mode_request_topic",
       server_node_config.dds_mode_request_topic);
-  get_parameter("dds_path_request_topic", 
+  get_parameter("dds_path_request_topic",
       server_node_config.dds_path_request_topic);
   get_parameter(
       "dds_destination_request_topic",
       server_node_config.dds_destination_request_topic);
-  get_parameter("update_state_frequency", 
+  get_parameter("update_state_frequency",
       server_node_config.update_state_frequency);
   get_parameter(
       "publish_state_frequency", server_node_config.publish_state_frequency);
@@ -148,7 +148,7 @@ void ServerNode::start(Fields _fields)
   // available
 
   update_state_callback_group = create_callback_group(
-      rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);
+      rclcpp::CallbackGroupType::MutuallyExclusive);
 
   update_state_timer = create_wall_timer(
       100ms, std::bind(&ServerNode::update_state_callback, this),
@@ -159,9 +159,9 @@ void ServerNode::start(Fields _fields)
   // handling requests from RMF to be sent down to the clients
 
   fleet_state_pub_callback_group = create_callback_group(
-      rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);
+      rclcpp::CallbackGroupType::MutuallyExclusive);
 
-  fleet_state_pub = 
+  fleet_state_pub =
       create_publisher<rmf_fleet_msgs::msg::FleetState>(
           server_node_config.fleet_state_topic, 10);
 
@@ -174,7 +174,7 @@ void ServerNode::start(Fields _fields)
   // Mode request handling
 
   auto mode_request_sub_opt = rclcpp::SubscriptionOptions();
-  
+
   mode_request_sub_opt.callback_group = fleet_state_pub_callback_group;
 
   mode_request_sub = create_subscription<rmf_fleet_msgs::msg::ModeRequest>(
@@ -207,7 +207,7 @@ void ServerNode::start(Fields _fields)
 
   destination_request_sub_opt.callback_group = fleet_state_pub_callback_group;
 
-  destination_request_sub = 
+  destination_request_sub =
       create_subscription<rmf_fleet_msgs::msg::DestinationRequest>(
           server_node_config.destination_request_topic, rclcpp::QoS(10),
           [&](rmf_fleet_msgs::msg::DestinationRequest::UniquePtr msg)
@@ -257,10 +257,10 @@ void ServerNode::transform_fleet_to_rmf(
   // RCLCPP_INFO(
   //     get_logger(), "    fleet->rmf scaled: (%.3f, %.3f)",
   //     scaled[0], scaled[1]);
-      
+
   _rmf_frame_location.x = scaled[0];
   _rmf_frame_location.y = scaled[1];
-  _rmf_frame_location.yaw = 
+  _rmf_frame_location.yaw =
       _fleet_frame_location.yaw - server_node_config.rotation;
 
   _rmf_frame_location.t = _fleet_frame_location.t;
@@ -273,8 +273,8 @@ void ServerNode::transform_rmf_to_fleet(
 {
   // It feels easier to read if each operation is a separate statement.
   // The compiler will be super smart and elide all these operations.
-  const Eigen::Vector2d scaled = 
-      server_node_config.scale * 
+  const Eigen::Vector2d scaled =
+      server_node_config.scale *
       Eigen::Vector2d(_rmf_frame_location.x, _rmf_frame_location.y);
 
   // RCLCPP_INFO(
@@ -283,13 +283,13 @@ void ServerNode::transform_rmf_to_fleet(
 
   const Eigen::Vector2d rotated =
       Eigen::Rotation2D<double>(server_node_config.rotation) * scaled;
-  
+
   // RCLCPP_INFO(
   //     get_logger(), "    rmf->fleet rotated: (%.3f, %.3f)",
   //     rotated[0], rotated[1]);
 
   const Eigen::Vector2d translated =
-      rotated + 
+      rotated +
       Eigen::Vector2d(
           server_node_config.translation_x, server_node_config.translation_y);
 
@@ -299,7 +299,7 @@ void ServerNode::transform_rmf_to_fleet(
 
   _fleet_frame_location.x = translated[0];
   _fleet_frame_location.y = translated[1];
-  _fleet_frame_location.yaw = 
+  _fleet_frame_location.yaw =
       _rmf_frame_location.yaw + server_node_config.rotation;
 
   _fleet_frame_location.t = _rmf_frame_location.t;
@@ -356,7 +356,8 @@ void ServerNode::update_state_callback()
     if (it == robot_states.end())
       RCLCPP_INFO(
           get_logger(),
-          "registered a new robot: " + ros_rs.name);
+          "registered a new robot: [%s]",
+          ros_rs.name.c_str());
 
     robot_states[ros_rs.name] = ros_rs;
   }
@@ -377,7 +378,7 @@ void ServerNode::publish_fleet_state()
     transform_fleet_to_rmf(fleet_frame_rs.location, rmf_frame_rs.location);
 
     // RCLCPP_INFO(
-    //     get_logger(), 
+    //     get_logger(),
     //     "robot location: (%.1f, %.1f, %.1f) -> (%.1f, %.1f, %.1f)",
     //     fleet_frame_rs.location.x,
     //     fleet_frame_rs.location.y,
@@ -385,13 +386,13 @@ void ServerNode::publish_fleet_state()
     //     rmf_frame_rs.location.x,
     //     rmf_frame_rs.location.y,
     //     rmf_frame_rs.location.yaw);
-    
+
     rmf_frame_rs.name = fleet_frame_rs.name;
     rmf_frame_rs.model = fleet_frame_rs.model;
     rmf_frame_rs.task_id = fleet_frame_rs.task_id;
     rmf_frame_rs.mode = fleet_frame_rs.mode;
     rmf_frame_rs.battery_percent = fleet_frame_rs.battery_percent;
-  
+
     rmf_frame_rs.path.clear();
     for (const auto& fleet_frame_path_loc : fleet_frame_rs.path)
     {

--- a/free_fleet_server_ros2/src/ServerNode.hpp
+++ b/free_fleet_server_ros2/src/ServerNode.hpp
@@ -76,16 +76,16 @@ private:
       const std::string& fleet_name, const std::string& robot_name);
 
   void transform_fleet_to_rmf(
-      const rmf_fleet_msgs::msg::Location& fleet_frame_location, 
+      const rmf_fleet_msgs::msg::Location& fleet_frame_location,
       rmf_fleet_msgs::msg::Location& rmf_frame_location) const;
 
   void transform_rmf_to_fleet(
-      const rmf_fleet_msgs::msg::Location& rmf_frame_location, 
+      const rmf_fleet_msgs::msg::Location& rmf_frame_location,
       rmf_fleet_msgs::msg::Location& fleet_frame_location) const;
 
   // --------------------------------------------------------------------------
 
-  rclcpp::Subscription<rmf_fleet_msgs::msg::ModeRequest>::SharedPtr 
+  rclcpp::Subscription<rmf_fleet_msgs::msg::ModeRequest>::SharedPtr
       mode_request_sub;
 
   void handle_mode_request(rmf_fleet_msgs::msg::ModeRequest::UniquePtr msg);
@@ -107,25 +107,25 @@ private:
 
   // --------------------------------------------------------------------------
 
-  rclcpp::callback_group::CallbackGroup::SharedPtr update_state_callback_group;
+  rclcpp::CallbackGroup::SharedPtr update_state_callback_group;
 
   rclcpp::TimerBase::SharedPtr update_state_timer;
 
   std::mutex robot_states_mutex;
 
-  std::unordered_map<std::string, rmf_fleet_msgs::msg::RobotState> 
+  std::unordered_map<std::string, rmf_fleet_msgs::msg::RobotState>
       robot_states;
 
   void update_state_callback();
 
   // --------------------------------------------------------------------------
 
-  rclcpp::callback_group::CallbackGroup::SharedPtr 
+  rclcpp::CallbackGroup::SharedPtr
       fleet_state_pub_callback_group;
 
   rclcpp::TimerBase::SharedPtr fleet_state_pub_timer;
 
-  rclcpp::Publisher<rmf_fleet_msgs::msg::FleetState>::SharedPtr 
+  rclcpp::Publisher<rmf_fleet_msgs::msg::FleetState>::SharedPtr
       fleet_state_pub;
 
   void publish_fleet_state();

--- a/free_fleet_server_ros2/src/main.cpp
+++ b/free_fleet_server_ros2/src/main.cpp
@@ -27,7 +27,7 @@ int main(int argc, char** argv)
   rclcpp::init(argc, argv);
   std::cout << "Greetings from free_fleet_server_ros2" << std::endl;
 
-  free_fleet::ros2::ServerNodeConfig server_node_config = 
+  free_fleet::ros2::ServerNodeConfig server_node_config =
       free_fleet::ros2::ServerNodeConfig::make();
   server_node_config.fleet_name = "free_fleet_server_ros2";
 
@@ -36,7 +36,7 @@ int main(int argc, char** argv)
     return 1;
 
   rclcpp::executors::MultiThreadedExecutor executor {
-      rclcpp::executor::ExecutorArgs(), 2};
+      rclcpp::ExecutorOptions(), 2};
   executor.add_node(server_node);
   executor.spin();
 

